doc/api/url.md

```diff
@@ -2,139 +2,244 @@
+of the `query` or `hash` components, delimited by either the ASCII question
+mark (`?`) or hash (`#`) characters.
 
-### Escaped Characters
+For example `'/p/a/t/h'`
 
-Spaces (`' '`) and the following characters will be automatically escaped in the
-properties of URL objects:
+No decoding of the path string is performed.
 
-```
-< > " ` \r \n \t { } | \ ^ '
-```
+### urlObject.search
+
+The `search` property consists of the entire "query string" portion of the
+URL, including the leading ASCII question mark (`?`) character.
+
+For example: `'?query=string'`
+
+### urlObject.path
+
+The `path` property is a concatenation of the `pathname` and `search`
+components.
+
+For example: `'/p/a/t/h?query=string'`
+
+No decoding of the `path` is performed.
+
+### urlObject.query
+
+The `query` property is either the "params" portion of the query string (
+everything *except* the leading ASCII question mark (`?`), or an object
+returned by the `querystring` module's `parse()` function:
+
+For example: `'query=string'` or `{'query': 'string'}`
 
----
+### urlObject.hash
 
-The following methods are provided by the URL module:
+The `hash` property consists of the "fragment" portion of the URL including
+the leading ASCII hash (`#`) character.
 
-## url.format(urlObj)
+For example: `'#hash'`
+
+## url.format(urlObject)
 <!-- YAML
 added: v0.1.25
 -->
 
-Take a parsed URL object, and return a formatted URL string.
-
-Here's how the formatting process works:
-
-* `href` will be ignored.
-* `path` will be ignored.
-* `protocol` is treated the same with or without the trailing `:` (colon).
-  * The protocols `http`, `https`, `ftp`, `gopher`, `file` will be
-    postfixed with `://` (colon-slash-slash) as long as `host`/`hostname` are present.
-  * All other protocols `mailto`, `xmpp`, `aim`, `sftp`, `foo`, etc will
-    be postfixed with `:` (colon).
-* `slashes` set to `true` if the protocol requires `://` (colon-slash-slash)
-  * Only needs to be set for protocols not previously listed as requiring
-    slashes, such as `mongodb://localhost:8000/`, or if `host`/`hostname` are absent.
-* `auth` will be used if present.
-* `hostname` will only be used if `host` is absent.
-* `port` will only be used if `host` is absent.
-* `host` will be used in place of `hostname` and `port`.
-* `pathname` is treated the same with or without the leading `/` (slash).
-* `query` (object; see `querystring`) will only be used if `search` is absent.
-* `search` will be used in place of `query`.
-  * It is treated the same with or without the leading `?` (question mark).
-* `hash` is treated the same with or without the leading `#` (pound sign, anchor).
+* `urlObject` {Object} A URL object (either as returned by `url.parse()` or
+  constructed otherwise).
+
+The `url.format()` method processes the given URL object and returns a formatted
+URL string.
+
+The formatting process essentially operates as follows:
```

Comment:
Is documenting this entire process *really* necessary? Would it be better/clearer to just document potential gotchas instead in each of the url component doc sections (e.g. like how node parses `user:pass:word` for the `auth` component)?